inc/saiswitch.h

```diff
@@ -1705,6 +1705,25 @@ typedef enum _sai_switch_attr_t
      */
     SAI_SWITCH_ATTR_UNINIT_DATA_PLANE_ON_REMOVAL,
 
+    /**
+     * @brief Instruct SAI to execute switch preshutdown
+     *
+     * Indicates controlled switch preshutdown as first step of warm shutdown.
+     * This hint is optional, SAI application could skip this step and
+     * go directly to warm shutdown.
+     * This hint should be ignored, if at the time SAI receives this hint,
+     * SAI_SWITCH_ATTR_RESTART_WARM is NOT already set to TRUE.
+     * The scope of preshutdown is to backup SAI/SDK data, but leave CPU port
```

Comment:
CPU port should get disabled as part of the detach sequence in remove_switch() API. So, Isnt the CPU packets will be processed until the remove_switch() is called in the current model already? 
Is this attribute meant to introduce an intermediate stage in the "planned restart" operation so that the backup process can be done? Please clarify.

Reply:
What you said is true. However, remove_switch() without pre-shutdown could take very long time. And we don't known what is the moment when cpu port is shutdown. So we couldn't time ourselves to send out these 'last' messages. With 2 stage shutdown, we can precisely control when these packet will be sent out, and knowing that the remove_switch() will be a faster operation.